### PR TITLE
feat: add hono adapter

### LINKED ISSUE #621: Support Hono
I've tried with all the available adapters - and none of them work with [Hono](https://hono.dev/). Hono is a barebones framework that strives to support as many runtimes by closely adhering to just web standards, and it's fast. I'm trying to build most stuff around it now - to keep the code easy to reuse and move around. But I'm using Bull a lot - so I'd really like to be able to spin up Bullboard on it as well.. :)

Is there an adapter for Hono in the works? Or is there a simpler (or more granular) way to set up Bullboard that might work?

### PR DESCRIPTION
**Pull request by nihalgonsalves**
Resolves #621

Add a HonoAdapter as well as an example using it. I modified and generalised this adapter from https://github.com/felixmosh/bull-board/issues/621#issuecomment-1835374208 by @abarakat4202 

Usage:

```ts
import { Hono } from 'hono';
import { createBullBoard } from '@bull-board/api';
import { HonoAdapter } from '@bull-board/hono';

// one of:
import { serveStatic } from 'hono/bun';
import { serveStatic } from 'hono/cloudflare-pages';
import { serveStatic } from 'hono/cloudflare-workers';
import { serveStatic } from 'hono/deno';
import { serveStatic } from '@hono/node-server/serve-static';

const app = new Hono();
const serverAdapter = new HonoAdapter(serveStatic);

createBullBoard({
  queues: [...],
  serverAdapter,
});

serverAdapter.setBasePath('/ui');
app.route('/ui', serverAdapter.registerPlugin());
```

### CODE REVIEW

File: examples/with-hono/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "description": "Example of how to use Hono server with bull-board",
+  "main": "index.js",
+  "scripts": {
+    "start": "node index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "felixmosh",
+  "license": "ISC",
+  "dependencies": {
+    "@bull-board/hono": "../../packages/hono",
```
Comment (nihalgonsalves) [inline examples/with-hono/package.json:13]: should be updated to the published version once published

**Comment by felixmosh** [comment]
Thank you for your contribution 🙏🏽

**Comment by felixmosh** [comment]
Hi @nihalgonsalves I've got a PR for updating Hono version, is that safe to update?

**Comment by nihalgonsalves** [comment]
@felixmosh sorry, was on vacation and missed that - I created #778 :)

**Comment by felixmosh** [comment]
Thank you, appriciate it 🙏🏽

**Pull request merged**